### DISCUSSION TITLE
Welcome to AWS-Lambda-Docker-Puppeteer Discussions!

### DISCUSSION THREAD

**Discussion by VikashLoomba** [VikashLoomba/AWS-Lambda-Docker#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you you're doing with this repo.

**Comment by carlreiser2** [comment]
How big is the docker image? I have something like this running in AWS but the image is 2.2gb 👎  which causes terrible cold start times.

**Reply by carlreiser2** [reply]
Also wondering if you've tried parallelization of browsers / pages in AWS and if thats possible with this implementation

**Reply by VikashLoomba** [reply]
Image is around .5GB with the current dependencies. Cold start times for me are reasonable but it can take a couple seconds. I would recommend using provisioned concurrency to eliminate cold start lag. 

The easy way to parallelize would be to run separate invocations of the lambda. If you want to constrain it to a single lambda, it would be possible to spin up multiple browser contexts in the function but you would have to be more conscious of memory constraints.

**Reply by VikashLoomba** [reply]
Just as an additional note, I would recommend Playwright over Puppeteer in lambda for parallel browsers in a single invocation. I found it easier to run Playwright with multiple pages/browsers over Puppeteer. I have a repo for Playwright Docker in Lambda [here](https://github.com/VikashLoomba/AWS-Lambda-Docker-Playwright).

**Reply by carlreiser2** [reply]
Interesting, I will probably try this implementation considering the image is much smaller. Provisioned concurrency is a good solution but it seems to me that AWS bumped the prices up. I am glad that container images are supported for lambda now, before trying this was a complete nightmare in my experience.

**Reply by carlreiser2** [reply]
Also, I have github actions setup which tags and pushes images to ECR based on version numbers. Then it connects the newly tagged image to the lambda so it ties it together nicely for continuous delivery. I think having a pipeline like that could be somewhat useful for this repo.

**Reply by VikashLoomba** [reply]
> Also, I have github actions setup which tags and pushes images to ECR based on version numbers. Then it connects the newly tagged image to the lambda so it ties it together nicely for continuous delivery. I think having a pipeline like that could be somewhat useful for this repo.

Just added this!

**Reply by VikashLoomba** [reply]
For the first one, make sure your user data dir on the disk is saved in /tmp. Thats the only place AWS Lambda supports writing files to during runtime. For the second issue, it seems as though the aws-lambda-ric npm package didn't install correctly for some reason.
